src/ext/conversion/__test__/exportProperties.spec.ts

```diff
@@ -0,0 +1,118 @@
+  describe('getColumnWidth', () => {
+    let colIdx;
+    const qDimensions = [
+      {
+        qDef: {
+          textAlign: {
+            auto: true,
+            align: 'left' as Align,
+          },
+          columnWidth: {
+            type: 'auto' as ColumnWidthType,
+            pixels: 200,
+            percentage: 20,
+          },
+        },
+        qAttributeExpressions: [],
+      },
+    ];
+    const qMeasures = [
+      {
+        qDef: {
+          textAlign: {
+            auto: true,
+            align: 'left' as Align,
+          },
+          columnWidth: {
+            type: 'pixels' as ColumnWidthType,
+            pixels: 200,
+            percentage: 20,
+          },
+        },
+        qAttributeExpressions: [],
+      },
+      {
+        qDef: {
+          textAlign: {
+            auto: true,
+            align: 'left' as Align,
+          },
+          columnWidth: {
+            type: 'FitToContent' as ColumnWidthType,
+            pixels: 200,
+            percentage: 20,
+          },
+        },
+        qAttributeExpressions: [],
+      },
+    ];
+
+    it('should get the correct columnWidth for the dimension', () => {
+      colIdx = 0;
+      const columnWidth = getColumnWidth(colIdx, qDimensions, qMeasures);
+      expect(columnWidth).toEqual(-1);
+    });
+
+    it('should get the correct columnWidth for the first measure', () => {
+      colIdx = 1;
+      const columnWidth = getColumnWidth(colIdx, qDimensions, qMeasures);
+      expect(columnWidth).toEqual(200);
+    });
+
+    it('should get the correct columnWidth for the second measure', () => {
+      colIdx = 2;
+      const columnWidth = getColumnWidth(colIdx, qDimensions, qMeasures);
+      expect(columnWidth).toEqual(-1);
+    });
+  });
+
+  describe('exportProperties', () => {
+    const propertyTree = {
+      qChildren: [],
+      qProperty: {
+        qHyperCubeDef: {
+          qDimensions: [
+            {
+              qDef: {
+                textAlign: {
+                  auto: true,
+                  align: 'left' as Align,
+                },
+                columnWidth: {
+                  type: 'auto' as ColumnWidthType,
+                  pixels: 200,
+                  percentage: 20,
+                },
+              },
+            },
+          ],
+          qMeasures: [
+            {
+              qDef: {
+                textAlign: {
+                  auto: true,
+                  align: 'left' as Align,
+                },
+                columnWidth: {
+                  type: 'pixels' as ColumnWidthType,
+                  pixels: 200,
+                  percentage: 20,
+                },
+              },
+            },
+          ],
+        },
+      },
+    } as unknown as PropTree;
+    const hyperCubePath = undefined;
+
+    const expFormat = exportProperties(propertyTree, hyperCubePath);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
why do you have to do this?

Reply:
cause in types, propertyTree.qProperty.qHyperCubeDef.qMeasures[0].qDef does not have columnWidth

Reply:
then can't we set the extended hypercube type on the propertyTree?

Reply:
that is in @types/qlik-engineapi and with a very deep hierarchy. I do not know how to do that yet.

Reply:
wouldn't that be just be setting qProperty to something like this:
```
export interface PropTree {
  qChildren: unknown[];
  qProperty: { qHyperCubeDef: QHyperCubeDef }
}
```

in types.ts?

Reply:
the ExportFormat one should this be the generic from engine API, since that is not the sn-table props, I think

Reply:
good idea, I gonna do that.

Reply:
Fixed